### PR TITLE
Added JUnit tests

### PR DESCRIPTION
**Pull request by VesnaGrumic**
Added JUnit tests for DefaultClient class
Added JUnit tests for DefaultApplication class
Added JUnit tests for DefaultData class

Signed-off-by: VesnaGrumic <vesna.grumic@endava.com>

**Related Issue**
/

**Description of the solution adopted**
/

**Screenshots**
/

**Any side note on the changes made**
/

### CODE REVIEW

**Review by LeoNerdoG — APPROVED** [review]
Hey @Coduz,
I have checked this PR and it looks OK. can you please emrge it after CI finishes? Thanks! 
P.S. Please add label "test" to the PR.

File: client/gateway/spi/src/test/java/org/eclipse/kapua/client/gateway/spi/DefaultClientTest.java [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+        final Channel channel = Mockito.mock(Channel.class);
+        final ScheduledExecutorService executorService = Mockito.mock(ScheduledExecutorService.class);
+        final Set<Module> modules = new HashSet<>();
+        modules.add(Mockito.mock(Module.class));
+        DefaultClient client = new DefaultClient(channel, executorService, modules);
+        assertEquals("Expected and actual values should be the same.", executorService, client.executor);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void defaultClientChannelNullTest() {
```
Comment (Coduz) [inline client/gateway/spi/src/test/java/org/eclipse/kapua/client/gateway/spi/DefaultClientTest.java:79]: Please check this.
The test is either expected to throw NPE or perform the last `.assertEquals`.
In the first case `assertEquals` should be removed and let just the NPE happen

**Review by Coduz — CHANGES_REQUESTED** [review]
Hi @VesnaGrumic 

I've made a comment on one of the test because it is not the correct way of testing throwing of exception. 

There are many places in which that is done, so please check 'em all!

Thanks

_Alberto_

**Comment by lorthirk** [comment]
Superseded by #3150

**Pull request closed without merging**